src/main/java/com/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieDao;
+import com.cinema.lib.Dao;
+import com.cinema.model.Movie;
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long movieId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(movieId);
```

Comment:
please see checklist: you don’t need to set id for your entity explicitly - it will be done by Hibernate :)

Reply:
🤦‍♀️